File: TwitterElectron/TwitterElectron/RendererProcess/TwitterCredentials.cs
using Bridge;

namespace TwitterElectron.RendererProcess
{
    [ObjectLiteral]
    public class TwitterCredentials
    {
        public string ApiKey { get; set; }

        public string ApiSecret { get; set; }

        public string AccessToken { get; set; }

        public string AccessTokenSecret { get; set; }
    }
}

File: TwitterElectron/TwitterElectron/App/TwitterCredentials.js
Bridge.assembly("TwitterElectron", function ($asm, globals) {
    "use strict";

    Bridge.define("TwitterElectron.RendererProcess.TwitterCredentials", {
        $literal: true,
        methods: {
            AreValid: function () {
                return !System.String.isNullOrEmpty(this.ApiKey) && !System.String.isNullOrEmpty(this.ApiSecret) && !System.String.isNullOrEmpty(this.AccessToken) && !System.String.isNullOrEmpty(this.AccessTokenSecret);
            }
        }
    });
});
